core/src/main/scala/kafka/server/DelegationTokenManager.scala

```diff
@@ -140,13 +140,13 @@ object DelegationTokenManager {
 
     val allow =
     //exclude tokens which are not requested
-      if (!owners.isEmpty && !owners.get.exists(owner => token.ownerOrRenewer(owner))) {
+      if (owners.isDefined && !owners.get.exists(owner => token.ownerOrRenewer(owner))) {
```

Comment:
although `isDefined` and `nonEmpty` are equivalent but I was wondering if there is a reason for using former at some places and the latter at others?

Reply:
`isDefined` and `nonEmpty` are equivalent on `Option`. On `String` there is not `isDefined`, so we need to use `nonEmpty`